flux-middle/src/const_eval.rs

```diff
@@ -0,0 +1,87 @@
+use flux_fixpoint::Sign::{Negative, Positive};
+use rustc_middle::ty::{self as rustc_ty, ParamEnv, TyCtxt, TyKind};
+
+use crate::rustc::mir::Constant;
+pub fn scalar_int_to_rty_constant<'tcx>(
+    tcx: TyCtxt<'tcx>,
+    scalar: rustc_ty::ScalarInt,
+    ty: rustc_middle::ty::Ty<'tcx>,
+) -> Option<crate::rty::Constant> {
+    match ty.kind() {
+        TyKind::Int(_) => {
+            let i = scalar_to_int(tcx, scalar, ty)?;
+            if 0 <= i {
+                Some(crate::rty::Constant::Int(Positive, i as u128))
+            } else {
+                Some(crate::rty::Constant::Int(Negative, (0 - i) as u128))
+            }
```

Comment:
Use `Constant::from(i)`
```suggestion
            Constant::from(i)
```

Reply:
ups, this was supposed to be part of the review.